docs/migrationstate.rst

```diff
@@ -0,0 +1,52 @@
+migrationstate
+==============
+
+:synopsis: Dumps/Restores the current migration state of all installed apps.
+
+
+Why?
+----
+
+If you, for instance, want to re-import data taken out of an existing
+database with the `dumpdata` command into a fresh one, you need to ensure
+the database tables of all apps are migrated to the state they were when
+the `dumpdata` command was run.
+
+To achieve this, you'd need to run::
+
+  $ ./manage.py showmigrations
+
+and manually apply all migrations that were applied before on the
+new system, which is annoying and susceptible to mistakes.
+
+That's why the `migrationstate` command was written. It automates this
+task for you by storing the current migration state in machine-readable
+JSON.
+
+
+How?
+----
+
+To dump the migration state of all apps installed in your project::
+
+  $ ./manage.py migrationstate dump > migrationstate.json
+
+To dump the migration state of just a single app (`appname`)::
+
+  $ ./manage.py migrationstate appname > migrationstate.json
+
+To apply the migration state stored in `migrationstate.json`::
+
+  $ ./manage.py migrationstate load < migrationstate.json
+
+
+Caveats
+-------
+
+Consistency of Migrations
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It should be clear, but I'm mentioning it anyway. To be able to reproduce
+a previously dumped migration state, all migration files up to that
+state still need to be present when restoring. Never delete or change
+migrations that have already been deployed.
```

Comment:
This concluded `squashing` the migrations I assume ? If so would be good to mention that as well.

Reply:
It'll work as long as the migrations having the stored names are present, so squashing is ok as long as the original migrations are kept as well.